### PR TITLE
Add upsert description to Update document API information

### PR DESCRIPTION
**Pull request by cwillum**
Signed-off-by: cwillum <cwmmoore@amazon.com>

### Description
Added information for `upsert` and `doc_as_upsert` in the _Update document_ section of REST API reference documentation.

### Issues Resolved
This addresses issue [#755](https://github.com/opensearch-project/documentation-website/issues/755).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwillum** [comment]
@tmccombs Thanks for getting this started for documentation. Let me know how things look and whether something more needs to be added.

File: _opensearch/rest-api/document-apis/update-document.md [hunk 1]
```diff
@@ -87,6 +87,35 @@ You can also use a script to tell OpenSearch how to update your document.
...
 
+### Upsert
+
+Upsert is an operation that conditionally updates an existing document or inserts a new one based on information in the object. In the sample below, the `upsert` object updates the name and adds the `age` field if a document already exists. If a document does not exist, a new one is indexed using content in the `upsert` object.
+
+```json
+{
+  "doc": {
+    "first_name": "Thomas",
+    "last_name": "Wayne"
```
Comment (Naarcha-AWS) [inline _opensearch/rest-api/document-apis/update-document.md:98]: Should the `last_name` is the "doc" body vs the `last_name` in the "upsert" body match? If not, why?

**Review by Naarcha-AWS — COMMENTED** [review]
Quick question for technical clarity, otherwise LGTM.

File: _opensearch/rest-api/document-apis/update-document.md (hunk 1 above)
Comment (cwillum) [reply _opensearch/rest-api/document-apis/update-document.md:98]: The idea in this example is that if the last name changes, the upsert updates the new information in an existing doc or adds (inserts) a new doc if it doesn't already exist. I used a completely different last name in the example to try making it clearer. I also worked some language in the description to help. Does that make better sense?

Comment (cwillum) [reply _opensearch/rest-api/document-apis/update-document.md]: Changed name completely to help make example explicit.

**Review by tmccombs — APPROVED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**